pkg/rpcserver/rpcserver.go

```diff
@@ -125,13 +135,16 @@ func New(cfg *mgrconfig.Config, mgr Manager, debug bool) (*Server, error) {
 	}, mgr)
 }
 
-func newImpl(ctx context.Context, cfg *Config, mgr Manager) (*Server, error) {
+func newImpl(ctx context.Context, cfg *Config, mgr Manager) (*server, error) {
```

Comment:
I wonder if it does really need to return `error` now -- the only case when it does should actually be ruled out by the `mgrconfig.Load` checks (?)

So I think we could just drop the `if sysTarget == nil` check as well as the `error` return value.